source/common/config/grpc_mux_subscription_impl.cc

```diff
@@ -90,6 +92,10 @@ std::string GrpcMuxSubscriptionImpl::resourceName(const ProtobufWkt::Any& resour
 }
 
 void GrpcMuxSubscriptionImpl::disableInitFetchTimeoutTimer() {
+  if (init_fetch_timer_) {
+    init_fetch_timer_->complete();
```

Comment:
This should probably be called only on successful config update i.e `onConfigUpdate() `and when it is rejected i.e. `Envoy::Config::ConfigUpdateFailureReason::UpdateRejected` case in  `onConfigUpdateFailed` method - so that it actually computes how much time the actual response took - if we complete it here it would be capped to `initial_fetch_timeout ` which is not correct - Envoy may continue to initialize after `initial_fetch_timeout `  but we want to measure actual time each xDS response takes right?